calyx-ir/src/guard.rs

```diff
@@ -418,8 +424,8 @@ where
     T: Eq,
 {
     /// updates self -> self & interval
-    pub fn add_interval(&mut self, interval: T) {
-        self.update(|g| g.and(Guard::Info(interval)));
+    pub fn add_interval(&mut self, timing_interval: T) {
```

Comment:
The `*_interval` naming convention is a little misleading here since the method is generic over `T` which represents the Info field of the underlying guard right? If this is only meant to be meaningful for StaticGuards, then restrict the `impl` to only static guards instead of being generic:
```
impl Guard<StaticTiming> { ... }
```